openmc/settings.py

```diff
@@ -1087,3 +1136,88 @@ def export_to_xml(self):
         tree = ET.ElementTree(self._settings_file)
         tree.write("settings.xml", xml_declaration=True,
                              encoding='utf-8', method="xml")
+
+
+class ResonanceScattering(object):
+    """Specification of the elastic scattering model for resonant isotopes.
+
+    Attributes
+    ----------
+    nuclide : openmc.nuclide.Nuclide
+        The nuclide affected by this resonance scattering treatment.
+    nuclide_0K : openmc.nuclide.Nuclide
+        This should be the same isotope as `nuclide`, but it should have an
+        `xs` attribute that identifies 0 Kelvin data.
+    method : str
+        The method used to sample outgoing scattering energies.  Valid options
+        are 'ARES', 'CXS' (constant cross section), 'DBRC' (Doppler broadening
+        rejection correction), and 'WCM' (weight correction method).
+    E_min : float
+        The minimum energy above which the specified method is applied.  By
+        default, CXS will be used below `E_min`.
+    E_max : float
+        The maximum energy below which the specified method is applied.  By
+        default, the asymptotic target-at-rest model is applied  above `E_max`.
+
+    """
+
+    def __init__(self):
+        self._nuclide = None
+        self._nuclide_0K = None
+        self._method = None
+        self._E_min = None
+        self._E_max = None
+
+    @property
+    def nuclide(self):
+        return self._nuclide
+
+    @property
+    def nuclide_0K(self):
+        return self._nuclide_0K
+
+    @property
+    def method(self):
+        return self._method
+
+    @property
+    def E_min(self):
+        return self._E_min
+
+    @property
+    def E_max(self):
+        return self._E_max
+
+    @nuclide.setter
+    def nuclide(self, nuc):
+        check_type('nuclide', nuc, Nuclide)
+        if nuc.zaid == None: raise ValueError("The `nuclide` must have an "
+             "explicitly defined `zaid` attribute.")
+        self._nuclide = nuc
+
+    @nuclide_0K.setter
+    def nuclide_0K(self, nuc):
+        check_type('nuclide_0K', nuc, Nuclide)
+        if nuc.zaid == None: raise ValueError("The `nuclide_0K` must have an "
+             "explicitly defined `zaid` attribute.")
+        self._nuclide_0K = nuc
+
+    @method.setter
+    def method(self, m):
+        check_type('method', m, basestring)
```

Comment:
This can be simplified as follows:

```python
@method.setter
def method(self, method):
    check_value('method', method, ('ARES', 'CXS', 'DBRC', 'WCM'))
    self._method = method
```